plugins/org.jkiss.dbeaver.data.gis.view/src/org/jkiss/dbeaver/ui/gis/panel/GISLeafletViewer.java

```diff
@@ -611,11 +631,13 @@ private void saveAttributeSettings() {
     }
 
     private void updateControlsVisibility() {
-        GC gc = new GC(browser.getDisplay());
-        try {
-            browser.execute("javascript:showTools(" + toolsVisible +");");
-        } finally {
-            gc.dispose();
+        if (browser != null) {
```

Comment:
Perhaps just

```java
        if (browser == null) {
            return;
        }
```

Would look nicer here

Reply:
I can't do return here, because I don't think that the following code not inside this if should be skipped in this case.

Reply:
Huh? But this PR just wraps the entire body of this method and surrounds it with `if (browser != null)` check 🤔

Reply:
Not the entire body. This method continues after line 157

Reply:
I was talking about the `updateControlsVisibility` method which is located at line 633. Are you sure we are referring to the same method?

Reply:
Oh, yes %) I can do as you ask